### PR TITLE
feat(frontend): add support for ci via drone

### PR DESCRIPTION
**Pull request by stephenliang**
This PR adds the ability for `apps` to depend on `frontend`. When `build` in `apps` is triggered, it will execute `prebuild` which then triggers a build of `frontend`.

### CODE REVIEW

File: lib/rake/build.rake [hunk 1]
```diff
@@ -186,7 +186,20 @@ namespace :build do
...
+      RakeUtils.yarn_install
+
+      # Only build frontend packages that are relevant to `code-dot-org/apps`
+      ChatClient.log 'Building <b>frontend</b>...'
+      RakeUtils.system "yarn release:dryrun --filter @code-dot-org/dsco "
+    end
+  end
+
   tasks = []
+  tasks << :frontend if CDO.build_apps
```
Comment (snickell) [inline lib/rake/build.rake]: Is order important? e.g. does frontend need to be built before apps? If order matters, I'd prefer a brief comment (not that we have them between apps and dashboard lol)

**Review by snickell — APPROVED** [review]
✅, looks good, thanks for integrating this.

**Review by kelbyhawn — APPROVED** [review]

File: lib/rake/build.rake (hunk 1 above)
Comment (stephenliang) [reply lib/rake/build.rake]: `frontend` does need to be built before `apps` since apps consumes the design system. I'll add in a comment.

File: lib/cdo/test_run_utils.rb [hunk 2]
```diff
@@ -97,4 +97,13 @@ def self.run_bin_tests
       end
     end
   end
+
+  def self.run_frontend_tests
```
Comment (cat5inthecradle) [inline lib/cdo/test_run_utils.rb:101]: the phrase `run_frontend_tests` really makes me want to find a new name for this application. "Frontend" is way too overloaded of a term.

Comment (snickell) [reply lib/cdo/test_run_utils.rb:101]: I DO want the name of this function to match the directory which is `code-dot-org/frontend`.

But, outside the scope of this PR, I agree with Darin that I think this reveals "frontend" to be too generic a name in general for what is, in practice today just: "the directory the homepage app lives in".

BTW, I would be totally supportive of @stephenliang responding: "fu guys stop nitpicking me / this is too late in the game for this kind of shitpost!!!".

As we've gotten further into this, I think its a fair question if mixing starting the CMS/marketing project with "starting a new general use directory for JS code called frontend/" is still the right move. I have a feeling we're 50% likely to end up in a world where future engineers are informed that "nobody remembers why the homepage app is under frontend/apps/marketing (and the design components it shares with dashboard), but that's just how it is".

I would love to have a time machine to go back and not name "dashboard" "dashboard".

I personally think "marketing" is sort of a confusing term too, and maybe this is just my age, but I be would "most delighted" as a new dev to find a directory called: `code-dot-org/homepage` with sub-dirs `design-system-storybook` and `contentful-cms`.

I don't at all mind the new setup and JS ideas (turbo etc) in frontend/, but I'm not sure that the homepage (which we want the MOST decoupled of anything) should live there.

As an 11th hour proposal (again, most likely stephen can and should ignore this unless it happens to resonate with him!):
1. cp -R frontend/ homepage/
2. rm -rf frontend/apps/*
3. git commit

Basically, keep frontend/ as a system for new dashboard-app-related js modules (currently empty), but make the homepage directory (along with the only coupling: design-system-storybook) very very clearly named and separated. 

Then, tying it back to this PR, you'd have:
```
def self.run_homepage_tests
```

Comment (stephenliang) [reply lib/cdo/test_run_utils.rb:101]: Names are quite difficult, I think it would make sense and I'd much recommend that we revisit names closer to the end of the project when the packages and code structures are in place. The code is currently structured to have the ability to quickly rename, if desired. I think once we know what everything does and how everything is pieced together, some clear names will emerge which we can address at that time. I'll create a task to circle back on this topic.

Comment (cat5inthecradle) [reply lib/cdo/test_run_utils.rb:101]: +1 to waiting, but also +1 to "homepage", I think it best captures the common language we use "When people visit the code.org homepage, they can click to log into code studio"

And a shoutout to "Pegasus" because meaningless fun names don't have this problem, so I'll cast a last minute vote to call this "Minotaur"
